src/bed.cpp

```diff
@@ -95,7 +95,7 @@ BedItem* BedItem::getNextBedItem() const
 
 bool BedItem::canUse(Player* player)
 {
-	if (!player || !house || !player->isPremium()) {
+	if (!player || !house || !player->isPremium() || player->getZone() != ZONE_PROTECTION) {
```

Comment:
house tile is automatically protection zone so this check is not needed

Reply:
This check is useless, yes. The player needs to be in PZ to use bed. In image above player can run from PVP if there is house without a wall.